net/net_compat.c

```diff
@@ -178,9 +178,13 @@ int getaddrinfo_retro(const char *node, const char *service,
    (void)in_addr;
    (void)info;
 
+#ifdef HAVE_LIBNX
+   if (!hints->ai_family || hints->ai_family == AF_UNSPEC)
```

Comment:
remove the `hints->ai_family == AF_UNSPEC` since AF_UNSPEC is defined as 0.
*Or* you can remove the ai_family check and keep the == AF_UNSPEC (which I personally prefer for this check)